src/schema/posts.ts

```diff
@@ -1218,6 +1254,46 @@ export const resolvers: IResolvers<any, Context> = {
 
       return data;
     },
+    relatedPosts: async (
+      _,
+      args: GQLPostRelationArgs,
+      ctx,
+      info,
+    ): Promise<ConnectionRelay<GQLPost>> => {
+      const post = await ctx.con
+        .getRepository(Post)
+        .findOneByOrFail({ id: args.id });
+      await ensureSourcePermissions(ctx, post.sourceId);
+
+      return queryPaginatedByDate(
+        ctx,
+        info,
+        args,
+        { key: 'createdAt' },
+        {
+          queryBuilder: (builder) => {
+            builder.queryBuilder = builder.queryBuilder
+              .leftJoin(
+                PostRelation,
+                'pr',
+                `pr."relatedPostId" = ${builder.alias}.id`,
```

Comment:
The naming is confusing, so just double-check that this is the right condition and not the other way around. Check the other PR where you actually add it to the database. Confirm that you use it the same way in both PRs

Reply:
There is a test that confirms it works.

Reply:
`postId` is owning side so in this case collection id.